scripts/screeps_tencent_batch_rl_runner.py

```
@@ -1119,6 +1130,138 @@ def safe_extract_tar(tar_path: Path, extract_dir: Path) -> None:
+    top_level_safety: dict[str, Any],
+    artifact_dir: Path,
+) -> dict[str, Any]:
+    iterations = policy_update_iterations(data.get("policyUpdateIterations"), "policyUpdateIterations")
+    safe_policy_update = validated_remote_policy_update(data.get("policyUpdate"), top_level_safety)
+    raw_artifact_path = data.get("policyUpdateArtifactPath")
+    update_iterations = 0
+    if isinstance(safe_policy_update, dict) and "iterations" in safe_policy_update:
+        update_iterations = policy_update_iterations(safe_policy_update.get("iterations"), "policyUpdate.iterations")
+    if iterations <= 0 and update_iterations <= 0:
+        if raw_artifact_path is not None:
+            raise BatchRunError("remote policyUpdateArtifactPath is present without positive policyUpdateIterations")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)